File: python_api/Context.py
from pyspark.sql import SparkSession

spark = SparkSession.builder.appName('app').getOrCreate()

workspaceDirectoryPath = "../../data_sets/workspace/spark/"
workspaceDirectoryPath_SL = "../../data_sets/workspace/scikit-learn/"
defaultDirectoryPath = "../../data_sets/default/"

fileName = "dataSet.csv"
dataFrameFileName = "dataFrame"
pipelineModelFileName = "pipelineModel"
textFeatureIndexerFileName = "textFeatureIndexer"
mlModelFileName = "mlModel"
oneHotEncoderFileName = "encoder"


##########################################################
def getDatSetFilePath_SL():
    return workspaceDirectoryPath_SL + fileName


def getDataFrameFilePath_SL():
    return workspaceDirectoryPath_SL + dataFrameFileName


def getMlModelFilePath_SL():
    return workspaceDirectoryPath_SL + mlModelFileName


def getOneHotEncoderFilePath_SL():
    return workspaceDirectoryPath_SL + oneHotEncoderFileName
##########################################################


def getDatSetFilePath():
    return workspaceDirectoryPath + fileName


def getDataFrameFilePath():
    return workspaceDirectoryPath + dataFrameFileName


def getPipelineModelFilePath():
    return workspaceDirectoryPath + pipelineModelFileName


def getTextFeatureIndexerFilePath():
    return workspaceDirectoryPath + textFeatureIndexerFileName

File: scala_api/src/main/scala/application/dataAnalysis/Context.scala
package application.dataAnalysis

import application.workspaceDirectoryPath
import org.apache.spark.sql.SparkSession

case object Context {

  val spark: SparkSession = SparkSession.builder()
    .appName("App")
    .config("spark.master", "local")
    .getOrCreate()

  def turnOfLoggingTerminal(): Unit = {
    import org.apache.log4j.{Level, Logger}
    val rootLogger = Logger.getRootLogger
    rootLogger.setLevel(Level.ERROR)
    Logger.getLogger("org.apache.spark").setLevel(Level.ERROR)
    Logger.getLogger("org.spark-project").setLevel(Level.ERROR)
  }

  turnOfLoggingTerminal()

  private val fileName: String = "dataSet.csv"
  private val dataFrameFileName: String = "dataFrame"
  private val modelFileName: String = "pipelineModel"
  private val textFeatureIndexerFileName: String = "textFeatureIndexer"

  val saveModels = true

  def getFileNamePath = workspaceDirectoryPath + fileName

  def getModelFilePath = workspaceDirectoryPath + modelFileName

  def getDataFilePath = workspaceDirectoryPath + dataFrameFileName

  def getTextFeatureIndexerFilePath = workspaceDirectoryPath + textFeatureIndexerFileName

}
